### ISSUE TITLE
DLS cooldown

### ISSUE THREAD

**Issue by aubergine10** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#583]
DLS is awesome, however it can sometimes lead to vehicles doing a lot of weaving.

Having a cooldown effect, that prevents multiple DLS in short succession, would alleviate that problem and also likely tackle the issue of vehicles changing lanes near outside connections.

It could work something like this:

* Vehicle decides to lane change due to DLS: set a flag indicating DLS is active (I assume there will already be something like that)
* When the DLS completes, set a counter or timer of some kind that defines either number of segments/nodes or amount of time before the vehicle can do another DLS
* As the vehicle continues along its route, the counter or timer decrements until <= 0 after which point it can do more DLS.

In terms of outside connections, vehicles already seem to start in pseudo-random lanes. So in theory it should just be a case of setting their anti-DLS counter/timer when they spawn at map edge to prevent them doing lane changes too soon after arriving on the map. See also: #100

**Comment by aubergine10** [comment]
Issues relating to lane changes at outside connections: #83, #100

**Comment by DaEgi01** [comment]
Yes, I also think that lane switches should come with a penalty and the AI should try to minimize them.
We must ensure that necessary lane switches to get the correct exit or turn are not affected by that.

**Comment by brunoais** [comment]
I think lane changing penalizing will be awful and create X-lane roads where only the (r/l)most lane is used and the other ones are left unused so that traffic doesn't change lanes.
I think it's better how it is now and, instead, use sensible numbers for DLS.

**Comment by kvakvs** [comment]
Store a counter field when the next lane change can be performed.
Use `Time.frameCount + COOLDOWN_CONST` to set the next allowed lane change.

**Comment by brunoais** [comment]
That is fine as long as there's an initial cooldown to allow lane changes from the entrance to the "fast lane"
